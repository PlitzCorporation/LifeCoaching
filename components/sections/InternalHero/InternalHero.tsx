import Image from 'next/image';
import React from 'react';

import altCreator from '@/helpers/altCreator';
import { WhiteThickLine } from '@/components/ui/decorations';

export interface InternalHeroProps {
	title: string;
	subtitle?: string;
	bgImage: string;
}

const InternalHero = (props: InternalHeroProps) => {
	const altText = altCreator(props.bgImage);

	return (
		<section className="relative h-96 lg:h-150">
			<div className="absolute w-full top-0 bottom-0 z-20">
				<div className="plitz-container flex flex-col justify-center items-center text-center relative h-full">
					<WhiteThickLine />
					<h1 className="text-white drop-shadow-plitz-darker mt-5 lg:mt-7 mb-7">
						{props.title}
					</h1>
					<h2 className="text-plitz-secondary drop-shadow-plitz-darker text-2xl lg:text-4xl text-balance">
						{props.subtitle}
					</h2>
				</div>
			</div>

			<div className="absolute w-full top-0 bottom-0 z-10 bg-gradient-to-b from-plitz-primary/50 to-plitz-body/90"></div>

			<div className="relative h-96 lg:absolute lg:h-150 w-full z-0">
				<Image
					src={
						!props.bgImage.includes('placeholder')
							? `${process.env.NEXT_PUBLIC_CMS}/${props.bgImage}`
							: `/images/big-placeholder.jpg`
					}
					alt={altText}
					width={0}
					height={0}
					sizes="100vw"
					priority
					className="object-cover object-bottom min-w-full min-h-full max-w-full max-h-full"
				/>
			</div>
		</section>
	);
};

export default InternalHero;
