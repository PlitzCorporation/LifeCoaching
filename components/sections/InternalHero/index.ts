import InternalHero from './InternalHero';

export default InternalHero;
