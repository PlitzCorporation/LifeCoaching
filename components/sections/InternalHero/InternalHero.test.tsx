import { render } from '@testing-library/react';
import InternalHero, { InternalHeroProps } from './InternalHero';

jest.mock('next/image', () => ({
	__esModule: true,
	// eslint-disable-next-line @typescript-eslint/no-explicit-any
	default: (props: any) => {
		// eslint-disable-next-line @next/next/no-img-element, jsx-a11y/alt-text
		return <img {...props} priority="true" />;
	},
}));

describe('InternalHero', () => {
	const OLD_ENV = process.env;

	beforeAll(() => {
		jest.resetModules();
		process.env.NEXT_PUBLIC_CMS = 'https://thedavid.plitz7.com';
	});

	afterAll(() => {
		process.env = OLD_ENV;
	});

	const props: InternalHeroProps = {
		title: 'Hero Title',
		subtitle: 'Hero Subtitle',
		bgImage: 'uploads/Reason4Hope/image.jpg',
	};

	it('renders the correct hero content', () => {
		const { container } = render(<InternalHero {...props} />);
		expect(container.firstChild).toMatchSnapshot();
	});
});
