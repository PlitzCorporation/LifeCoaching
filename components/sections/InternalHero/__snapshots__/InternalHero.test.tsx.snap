// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InternalHero renders the correct hero content 1`] = `
<section
  class="relative h-96 lg:h-150"
>
  <div
    class="absolute w-full top-0 bottom-0 z-20"
  >
    <div
      class="plitz-container flex flex-col justify-center items-center text-center relative h-full"
    >
      <div
        class="block w-32 h-1 bg-gradient-to-r from-white/5 to-white/100"
      />
      <h1
        class="text-white drop-shadow-plitz-darker mt-5 lg:mt-7 mb-7"
      >
        Hero Title
      </h1>
      <h2
        class="text-plitz-secondary drop-shadow-plitz-darker text-2xl lg:text-4xl text-balance"
      >
        Hero Subtitle
      </h2>
    </div>
  </div>
  <div
    class="absolute w-full top-0 bottom-0 z-10 bg-gradient-to-b from-plitz-primary/50 to-plitz-body/90"
  />
  <div
    class="relative h-96 lg:absolute lg:h-150 w-full z-0"
  >
    <img
      alt="uploads/Reason4Hope/image"
      class="object-cover object-bottom min-w-full min-h-full max-w-full max-h-full"
      height="0"
      priority="true"
      sizes="100vw"
      src="https://thedavid.plitz7.com/uploads/Reason4Hope/image.jpg"
      width="0"
    />
  </div>
</section>
`;
